index.html

```diff
@@ -101,19 +207,633 @@ <h3>
+          <li>If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+        not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets=]", throw a
+        "{{NotAllowedError}}" {{DOMException}}.
+          <li>If |options|["{{UDPSocketOptions/sendBufferSize}}"] is equal to 0, throw a {{TypeError}}.
+          <li>If |options|["{{UDPSocketOptions/receiveBufferSize}}"] is equal to 0, throw a {{TypeError}}.
+          <li>If |options|["{{UDPSocketOptions/readableStreamBufferSize}}"] is equal to 0, throw a {{TypeError}}.
+          
+          <li>Perform the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to open the TCP socket using the given |remoteAddress| and |remotePort| and the 
+                connection parameters (or their defaults) specified in |options|.
+              <li>
+                If this fails for any reason, [=queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>[=Reject=] the {{UDPSocket/[[connectionPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
+                  <li>Detach {{UDPSocket/[[closedPromise]]}}.
+                </ol>
+              <li>
+                On success, [=queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>[=initialize udp readable stream|Initialize=] {{UDPSocket/[[readable]]}}.
+                  <li>[=initialize udp writable stream|Initialize=] {{UDPSocket/[[writable]]}}.
+                  <li>Let |conn:UDPSocketConnection| be a new {{UDPSocketConnection}}.
+                  <li>Set |conn|["{{UDPSocketConnection/readable}}"] to [=this=].{{UDPSocket/[[readable]]}}.
+                  <li>Set |conn|["{{UDPSocketConnection/writable}}"] to [=this=].{{UDPSocket/[[writable]]}}.
+                  <li>Populate the remaining fields of |conn| using the information provided by the operating system:
+                      |conn|["{{UDPSocketConnection/remoteAddress}}"], |conn|["{{UDPSocketConnection/remotePort}}"],
+                      |conn|["{{UDPSocketConnection/localAddress}}"] and |conn|["{{UDPSocketConnection/localPort}}"].
+                  <li>Resolve [=this=].{{UDPSocket/[[connectionPromise]]}} with |conn|.
+                </ol>
+            </ol>
+        </ol>
+
+        <section data-dfn-for="UDPSocketOptions">
+          <h4><dfn>UDPSocketOptions</dfn> dictionary</h4>
+
+          <pre class="idl">
+            dictionary UDPSocketOptions {
+              [EnforceRange] unsigned long sendBufferSize;
+              [EnforceRange] unsigned long receiveBufferSize;
+
+              [EnforceRange] unsigned short readableStreamBufferSize;
```

Comment:
Yes, that was the original intention! Are you suggesting to require all high watermarks to be implementation-defined? For instance, right now there's an option to control the `ReadableStream` queue size for UDP (but it's not exactly important, imo).

Reply:
Yes, I think we should say that the high watermark is implementation-defined and we can pick a reasonable value for UDP the same as we do (implicitly because of the default Mojo data pipe size) for TCP.

Reply:
Fixed!